### PR TITLE
ansible-doc: restore role attributes

### LINKED ISSUE #82639: Regression: ansible-doc no longer shows attributes for roles
### Summary

This was removed as part of https://github.com/ansible/ansible/commit/bec27fb4c0a40c5f8bbcf26a475704227d65ee73#diff-9732216f79978750789dbeb4c2ed1280d841c4211306ecb1b5c70e3db3fcb2f7L1198.

I guess this was a mistake.

### Issue Type

Bug Report

### Component Name

ansible-doc

### Ansible Version

```console
latest devel
```

### Configuration

```console
-
```

### OS / Environment

-

### Steps to Reproduce

-

### Expected Results

-

### Actual Results

```console
-
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #82639.

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

**Comment by sivel** [comment]
@felixfontein we've had a discussion around re-adding this unexpected feature.  Since we didn't have any documentation explaining that unintended features could be removed without deprecation, we would be willing to potentially re-accept this.  

However, the stipulation being that, it would be merged with an immediate deprecation for future removal.

With the above being said, do you feel it would be useful to re-add this feature to start immediately preparing to stop using it, or should we just leave it removed?

**Comment by felixfontein** [comment]
@sivel while I still disagree that it should be removed/deprecated at all, having a proper deprecation period is better than just removing it.

**Comment by s-hertel** [comment]
I added a deprecation warning so this can be merged, hopefully this looks okay:

> [DEPRECATION WARNING]: The role testns.testcol.testrole's argument spec alternate contains the key "attributes", which will not be displayed by ansible-doc in the future. This was unintentionally allowed when plugin attributes were 
added, but the feature does not map well to role argument specs. This feature will be removed from ansible-core in version 2.20. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.

**Comment by felixfontein** [comment]
This also does need a deprecation docs fragment now. @s-hertel do you want to add that as well, or should I?

**Comment by s-hertel** [comment]
Since there was no changelog/documentation for the accidental feature, I didn't think it needed one, but if you want to add one feel free, I'm out until Monday.

File: changelogs/fragments/82678-role-entrypoint-attributes.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+deprecated_features:
+  - "ansible-doc - role entrypoint attributes are deprecated and will no longer be shown in ansible-doc from ansible-core 2.20 on (https://github.com/ansible/ansible/issues/82639, https://github.com/ansible/ansible/pull/82678)."
```
Comment (bcoca) [inline changelogs/fragments/82678-role-entrypoint-attributes.yml]: 'and eventually' .. right now this PR restores them doesn't it?

Comment (felixfontein) [reply changelogs/fragments/82678-role-entrypoint-attributes.yml]: Do you mean like
```suggestion
  - "ansible-doc - role entrypoint attributes are deprecated and eventually will no longer be shown in ansible-doc from ansible-core 2.20 on (https://github.com/ansible/ansible/issues/82639, https://github.com/ansible/ansible/pull/82678)."
```
?

Comment (bcoca) [reply changelogs/fragments/82678-role-entrypoint-attributes.yml]: even though you specified the version, previous phrasing could confuse people to think it was immediate, thanks!

File: lib/ansible/cli/doc.py [hunk 2]
```diff
@@ -1306,6 +1306,24 @@ def get_role_man_text(self, role, role_json):
...
+                display.deprecated(
+                    f'The role {role}\'s argument spec {entry_point} contains the key "attributes", '
+                    'which will not be displayed by ansible-doc in the future. '
+                    'This was unintentionally allowed when plugin attributes were added, '
+                    'but the feature does not map well to role argument specs.',
+                    version='2.20',
+                    collection_name='ansible.builtin',
+                )
+                text.append("")
+                text.append(_format("ATTRIBUTES:", 'bold'))
```
Comment (bcoca) [inline lib/ansible/cli/doc.py:1319]: I wonder if we should add (DEPRECATED) here to ensure people see it

Comment (felixfontein) [reply lib/ansible/cli/doc.py:1319]: Users could misunderstand that that the attributes shown are deprecated (and not that showing attributes is deprecated).

Comment (bcoca) [reply lib/ansible/cli/doc.py:1319]: good point, then we just need to rely on runtime message + release notes

**Review by bcoca — APPROVED** [review]

**Comment by felixfontein** [comment]
@s-hertel @bcoca thanks for reviewing and merging!

**Pull request merged**